packages/dd-trace/src/config.js

```diff
@@ -321,201 +227,74 @@ class Config {
       false
     )
     const DD_API_SECURITY_REQUEST_SAMPLE_RATE = coalesce(
-      appsec?.apiSecurity?.requestSampling,
+      this.appsecOpt?.apiSecurity?.requestSampling,
       parseFloat(process.env.DD_API_SECURITY_REQUEST_SAMPLE_RATE),
       0.1
     )
 
-    const remoteConfigOptions = options.remoteConfig || {}
-    const DD_REMOTE_CONFIGURATION_ENABLED = coalesce(
+    this.DD_REMOTE_CONFIGURATION_ENABLED = coalesce(
       process.env.DD_REMOTE_CONFIGURATION_ENABLED && isTrue(process.env.DD_REMOTE_CONFIGURATION_ENABLED),
       !inServerlessEnvironment
     )
-    const DD_REMOTE_CONFIG_POLL_INTERVAL_SECONDS = coalesce(
-      parseFloat(remoteConfigOptions.pollInterval),
-      parseFloat(process.env.DD_REMOTE_CONFIG_POLL_INTERVAL_SECONDS),
-      5 // seconds
-    )
 
-    const iastOptions = options?.experimental?.iast
-    const DD_IAST_ENABLED = coalesce(
-      iastOptions &&
-      (iastOptions === true || iastOptions.enabled === true),
-      process.env.DD_IAST_ENABLED,
-      false
-    )
-    const DD_TELEMETRY_LOG_COLLECTION_ENABLED = coalesce(
-      process.env.DD_TELEMETRY_LOG_COLLECTION_ENABLED,
-      DD_IAST_ENABLED
-    )
-
-    const DD_TELEMETRY_DEPENDENCY_COLLECTION_ENABLED = coalesce(
-      process.env.DD_TELEMETRY_DEPENDENCY_COLLECTION_ENABLED,
-      true
-    )
-
-    const defaultIastRequestSampling = 30
-    const iastRequestSampling = coalesce(
-      parseInt(iastOptions?.requestSampling),
-      parseInt(process.env.DD_IAST_REQUEST_SAMPLING),
-      defaultIastRequestSampling
-    )
-    const DD_IAST_REQUEST_SAMPLING = iastRequestSampling < 0 ||
-      iastRequestSampling > 100 ? defaultIastRequestSampling : iastRequestSampling
-
-    const DD_IAST_MAX_CONCURRENT_REQUESTS = coalesce(
-      parseInt(iastOptions?.maxConcurrentRequests),
-      parseInt(process.env.DD_IAST_MAX_CONCURRENT_REQUESTS),
-      2
-    )
-
-    const DD_IAST_MAX_CONTEXT_OPERATIONS = coalesce(
-      parseInt(iastOptions?.maxContextOperations),
-      parseInt(process.env.DD_IAST_MAX_CONTEXT_OPERATIONS),
-      2
-    )
-
-    const DD_IAST_DEDUPLICATION_ENABLED = coalesce(
-      iastOptions?.deduplicationEnabled,
-      process.env.DD_IAST_DEDUPLICATION_ENABLED && isTrue(process.env.DD_IAST_DEDUPLICATION_ENABLED),
-      true
-    )
-
-    const DD_IAST_REDACTION_ENABLED = coalesce(
-      iastOptions?.redactionEnabled,
-      !isFalse(process.env.DD_IAST_REDACTION_ENABLED),
-      true
-    )
+    this.iastOptions = options?.experimental?.iast
 
     const DD_IAST_REDACTION_NAME_PATTERN = coalesce(
-      iastOptions?.redactionNamePattern,
+      this.iastOptions?.redactionNamePattern,
       process.env.DD_IAST_REDACTION_NAME_PATTERN,
       null
     )
 
     const DD_IAST_REDACTION_VALUE_PATTERN = coalesce(
-      iastOptions?.redactionValuePattern,
+      this.iastOptions?.redactionValuePattern,
       process.env.DD_IAST_REDACTION_VALUE_PATTERN,
       null
     )
```

Comment:
Why are these two iast properties (`DD_IAST_REDACTION_NAME_PATTERN` and `DD_IAST_REDACTION_VALUE_PATTERN`) kept here and not treated like the rest applying option, env, etc?

Reply:
They are still used to set [iast.redactionNamePattern and iast.redactionValuePattern](https://github.com/DataDog/dd-trace-js/blob/7ce05bcaec3c5bf5f1e5168d4c042793829babb4/packages/dd-trace/src/config.js#L368-L371) in the old way, since they are not reported by telemetry. This PR only makes the changes necessary to report config settings to telemetry

Reply:
Ok but I asked to the iast members and we don't know why they shouldn't be sent via telemetry :sweat_smile:

Reply:
Not sure either, just asked in #instrumentation-telemetry

Reply:
The analytics team added [mappings](https://github.com/DataDog/dd-go/pull/109295) for iast.redactionNamePattern and iast.redactionValuePattern so backend can recognize them. I've updated the PR to include these two config values

Reply:
thank you!!